### PR TITLE
rework permissions into separate APIs for linux and windows

### PR DESCRIPTION
**Pull request by mwrock**
This depends on https://github.com/habitat-sh/core/pull/70

Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
fixes #5726

File: components/pkg-export-docker/src/docker.rs [hunk 1]
```diff
@@ -382,7 +376,16 @@ impl DockerBuildRoot {
         self.build_docker_image(ui, naming)
     }
 
+    #[cfg(windows)]
+    fn add_users_and_groups(&self, _ui: &mut UI) -> Result<()> {
```
[the remark this answers is not in the document]
Comment (baumanj) [reply components/pkg-export-docker/src/docker.rs]: Yeah, I thought that might be the case. For stuff like that, I'd suggest a windows implementation that contains `unreachable!()` or `unimplemented!()`. That way they can be defined in a way that makes the compiler happy, but ensures that the flow avoids actually calling them, which would be misleading.

File: components/pkg-export-tar/src/rootfs.rs [hunk 2]
```diff
@@ -25,12 +22,25 @@ use error::Result;
...
+
     let root = root.as_ref();
     fs::create_dir_all(root)?;
-    filesystem::chmod(root.to_str().unwrap(), 0o0750)?;
+    util::posix_perm::set_permissions(root.to_str().unwrap(), 0o0750)?;
+    Ok(())
+}
+
+#[cfg(windows)]
+pub fn create<T>(_root: T) -> Result<()>
```
[the remark this answers is not in the document]
Comment (mwrock) [reply components/pkg-export-tar/src/rootfs.rs]: I made the other changes but not sure how to gracefuly get around this one here. I could have the entire rootfs mod only compile on linux and split the calling `prepare_rootfs` into a windows and a linux version but both would be identical with the exception of that one line. So keeping the current scenario here seems a lesser evil but totally open to suggestions.

[the remark this answers is not in the document]
Comment (mwrock) [reply components/pkg-export-tar/src/rootfs.rs]: ah yes `unimplemented!`

**Comment by mwrock** [comment]
@baumanj are you good with the changes here? If so I'll merge the core side, repoint the cargo.lock here to core master and merge.

**Comment by baumanj** [comment]
Yes! Sorry, I missed that you had but fixes in already.

**Pull request merged**